skills/declarative/calendarSkill/dialogs/SearchEventsDialog/SearchEventsDialog.dialog

```diff
@@ -0,0 +1,267 @@
+                  "value": "=@@personName"
+                },
+                {
+                  "property": "dialog.type",
+                  "value": "= dialog.datetime.type"
+                }
+              ]
+            }
+          ],
+          "elseActions": []
+        },
+        {
+          "$kind": "Microsoft.Graph.Calendar.GetEvents",
+          "$designer": {
+            "id": "QXHo4q"
+          },
+          "resultProperty": "dialog.events",
+          "token": "=user.token.token",
+          "titleProperty": "=coalesce(dialog.title, null)",
+          "locationProperty": "=coalesce(dialog.location, null)",
+          "attendeesProperty": "=coalesce(dialog.personNameArray, null)",
+          "endProperty": "=coalesce(dialog.end, null)",
+          "timeZoneProperty": "=user.workingHours.timeZone.name",
+          "isFutureProperty": "=coalesce(dialog.futureEventsOnly, false)",
+          "isFutureEventsOnlyProperty": "=dialog.futureEventsOnly",
+          "dateTimeTypeProperty": "= coalesce(dialog.type, \"date\")",
+          "startProperty": "=coalesce(dialog.start, formatDateTime(utcNow(), 'yyyy-MM-dd'))"
+        },
+        {
+          "$kind": "Microsoft.IfCondition",
+          "$designer": {
+            "id": "hmq2AT"
+          },
+          "condition": "dialog.events != null && count(dialog.events) > 0",
+          "elseActions": [
+            {
+              "$kind": "Microsoft.IfCondition",
+              "$designer": {
+                "id": "l8Fufv"
+              },
+              "condition": "dialog.turnCount == 0",
+              "actions": [
+                {
+                  "$kind": "Microsoft.RepeatDialog",
+                  "$designer": {
+                    "id": "jOlhgq"
+                  },
+                  "activityProcessed": true,
+                  "options": {
+                    "doSelection": true,
+                    "futureEventsOnly": true,
+                    "turnCount": 1
+                  }
+                }
+              ],
+              "elseActions": [
+                {
+                  "$kind": "Microsoft.EndDialog",
+                  "$designer": {
+                    "id": "KDD3hl"
+                  }
+                }
+              ]
+            }
+          ],
+          "actions": [
+            {
+              "$kind": "Microsoft.Foreach",
+              "$designer": {
+                "id": "JFoghi"
+              },
+              "index": "dialog.foreach.index",
+              "value": "dialog.foreach.value",
+              "itemsProperty": "dialog.events",
+              "actions": [
+                {
+                  "$kind": "Microsoft.SetProperties",
+                  "$designer": {
+                    "id": "M5nVxN"
+                  },
+                  "assignments": [
+                    {
+                      "property": "dialog.events[dialog.foreach.index].Location",
```

Comment:
I think we can make this change in the getEvents or the sortEvents action to remove the need for this. I'll make this change and make a PR

Reply:
Actually, since I don't have your getEvents action changes, I'll let you make the change. Here is the method you can use to take the raw event payload and convert it into the schema I used for the card:


        private List<object> ParseEvents(DateTime currentDateTime, List<Event> events)
        {
            var eventsList = new List<object>();
            var i = 0;

            foreach (var ev in events)
            {
                var start = DateTime.Parse(ev.Start.DateTime);
                var end = DateTime.Parse(ev.End.DateTime);
                var duration = end.Subtract(start);
                var isCurrentEvent = false;

                if(start <= currentDateTime && currentDateTime <= end
                    || start.AddMinutes(-30) <= currentDateTime && currentDateTime <= start)
                {
                    // If event is currently ongoing, or will start in the next 30 minutes
                    isCurrentEvent = true;
                }

                eventsList.Add(new
                {
                    Id = i++,
                    ev.Subject,
                    ev.Start,
                    ev.End,
                    ev.Attendees,
                    ev.IsOnlineMeeting,
                    ev.OnlineMeeting,
                    Description = ev.BodyPreview,
                    Location = !string.IsNullOrEmpty(ev.Location.DisplayName) ? ev.Location.DisplayName : string.Empty,
                    DurationDays = duration.Days,
                    DurationHours = duration.Hours,
                    DurationMinutes = duration.Minutes,
                    isRecurring = ev.Type == EventType.Occurrence || ev.Type == EventType.SeriesMaster ? true : false,
                    isCurrentEvent,
                    ev.IsOrganizer,
                    ev.ResponseStatus.Response,
                    ev.Organizer
                });
            }

            return eventsList;
        }

Reply:
GetEvent is called before SortEvent, if I change the output of GetEvent, something will be wrong with SortEvent and maybe other parts. So I changed the SortEvent to handle two scenarios, one returns events grouped by date and the other returns sorted events directly. Both convert the event to the schema used in card.